components/UserNameForm.vue

```diff
@@ -0,0 +1,41 @@
+<script setup lang="ts">
+
+defineProps<{
+  placeholder?: string
+}>()
+
+defineEmits(['submit'])
+
+const usernameModel = defineModel<string>()
+</script>
+
+<template>
+  <form
+    class="grid grid-cols-[1fr_auto] gap-2 mx-auto border-black border-2 rounded-md shadow-sm leading-none w-full overflow-hidden"
+    @submit.prevent="$emit('submit')"
+  >
```

Comment:
Sorry to spot this only just now, but I think we should likely tweak the CSS here as we now have a bit too much vertical height on the form owing to too much padding on username form (current site on right).

![CleanShot 2023-09-15 at 10 48 40](https://github.com/danielroe/firstcommit.is/assets/28706372/2fa8097e-cfa3-4aae-b9ee-b5cd8a85b75d)

It may be that extracting the component is a too-early abstraction for this particular site ...

Reply:
No probs! I just committed a fix that changed the button margin - and in turn the height, so check that out if it looks good to you! Whatever feels best though, nothing is sacred 👍 If it's a case of premature optimisation it's no point losing sleep over this 😄

Reply:
Ah. But I wanted the home page one to have more space.

Reply:
Aaah, got it! It should look as expected now though 😄